File: app/src/main/java/com/example/workoutapp/WorkoutAdapter.java
package com.example.workoutapp;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

import java.util.ArrayList;
//Adapter class that works with WorkoutAdapter.ViewHolder to create a recycler view for the list of all workouts
public class WorkoutAdapter extends RecyclerView.Adapter<WorkoutAdapter.ViewHolder> {
    private final Context context;
    private final RecyclerViewInterface recyclerViewInterface;
    private final ArrayList<Workout> workouts;
    private DatabaseReference databaseReference;

    //Constructor and data
    public WorkoutAdapter(Context context, ArrayList<Workout> workouts, RecyclerViewInterface recyclerViewInterface) {
        this.context = context;
        this.workouts = workouts;
        this.recyclerViewInterface = recyclerViewInterface;
    }



    @NonNull
    @Override
    public WorkoutAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        //Creates view of each individual workout in the list
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.workout_card, parent, false);
        return new ViewHolder(view, recyclerViewInterface);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        //Assembles the array of workouts
        Workout workout = workouts.get(position);
        holder.title.setText(workout.getTitle());
        holder.content.setText(workout.getContent());
        databaseReference = FirebaseDatabase.getInstance().getReference();

    }

    @Override
    public int getItemCount() {
        return workouts.size();
    } //Gets number of workouts

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public TextView title;
        public TextView content;

        public ViewHolder(@NonNull View itemView, RecyclerViewInterface recyclerViewInterface) {
            //Creates the recyclerview scrollable list of workouts.
            super(itemView);
            title = itemView.findViewById(R.id.workout_title);
            content = itemView.findViewById(R.id.workout_content);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if(recyclerViewInterface != null){
                        int pos = getAbsoluteAdapterPosition();

                        if(pos != RecyclerView.NO_POSITION){
                            recyclerViewInterface.onItemClick(pos);
                        }
                    }
                }
            });
        }
    }
}

File: app/src/main/java/com/example/workoutapp/WorkoutAdapter.kt
package com.example.workoutapp

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase

//Adapter class that works with WorkoutAdapter.ViewHolder to create a recycler view for the list of all workouts
class WorkoutAdapter //Constructor and data
    (
    private val context: Context,
    private val workouts: ArrayList<Workout>,
    private val recyclerViewInterface: RecyclerViewInterface
) : RecyclerView.Adapter<WorkoutAdapter.ViewHolder>() {
    private var databaseReference: DatabaseReference? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        //Creates view of each individual workout in the list
        val view = LayoutInflater.from(parent.context).inflate(R.layout.workout_card, parent, false)
        return ViewHolder(view, recyclerViewInterface)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        //Assembles the array of workouts
        val workout = workouts[position]
        holder.title.text = workout.title
        holder.content.text = workout.content
        databaseReference = FirebaseDatabase.getInstance().getReference()
    }

    override fun getItemCount(): Int {
        return workouts.size
    } //Gets number of workouts

    class ViewHolder(itemView: View, recyclerViewInterface: RecyclerViewInterface?) :
        RecyclerView.ViewHolder(itemView) {
        var title: TextView
        var content: TextView

        init {
            //Creates the recyclerview scrollable list of workouts.
            title = itemView.findViewById(R.id.workout_title)
            content = itemView.findViewById(R.id.workout_content)
            itemView.setOnClickListener {
                if (recyclerViewInterface != null) {
                    val pos = getAbsoluteAdapterPosition()
                    if (pos != RecyclerView.NO_POSITION) {
                        recyclerViewInterface.onItemClick(pos)
                    }
                }
            }
        }
    }
}